libsplinter/src/rest_api/mod.rs

```diff
@@ -330,6 +330,93 @@ impl RequestGuard for Box<dyn RequestGuard> {
     }
 }
 
+/// Guards requests based on a minimum protocol version.
+///
+/// A protocol version is specified via the HTTP header `"X-SplinterProtocolVersion"`.  This header
+/// is a positive integer value.
+#[derive(Clone)]
+pub struct ProtocolVersionRangeGuard {
+    min: u32,
+    max: u32,
+}
+
+impl ProtocolVersionRangeGuard {
+    /// Constructs a new protocol version guard with the given minimum.
+    pub fn new(min: u32, max: u32) -> Self {
+        Self { min, max }
+    }
+}
+
+impl RequestGuard for ProtocolVersionRangeGuard {
+    fn evaluate(&self, req: &HttpRequest) -> Continuation {
+        if let Some(header_value) = req.headers().get("X-SplinterProtocolVersion") {
```

Comment:
May not want to include the `X-` in the header name: https://stackoverflow.com/a/3561399

Reply:
Good catch, will update

Reply:
Good catch, will update

Reply:
Good catch, will update

Reply:
Good catch, will update

Reply:
Fixed